agent/consul/server.go

```diff
@@ -880,6 +880,8 @@ func (s *Server) Leave() error {
 		}
 	}
 
+	s.raft.Leave()
```

Comment:
this seems like a pretty big thing, is it possible to add a test for this, so the regression can't happen in the future and prove the fix?